arcsi/api/utils.py

```diff
@@ -89,23 +99,24 @@ def broadcast_audio(archive_base, archive_idx, broadcast_file_name, broadcast_pl
 
 
 def process(archive_base, archive_idx, archive_file, archive_name):
-    archive_file_name = form_filename(archive_file, archive_name)
-    if archive_file_name != "":
-        archive_file_path = media_path(
-            archive_base, str(archive_idx), archive_file_name
-        )
-        archive_file.save(archive_file_path)
-        return archive_file_name
-    else:
+    if not allowed_file(archive_file):
         return None
```

Comment:
I was wondering where the corresponding error handling was. 

Here? 

https://github.com/mmmnmnm/lahmacun_arcsi/blob/024ba5ff1ecd0e773c4c5dde3068b09cc6b7e9dc/arcsi/api/item.py#L342-L349

Will a `None` return value be a Boolean `False` here? Btw, shouldn't it be `if not (image_file_name and not item.play_file_name)` (because we need both audio and image for a valid request)?

Reply:
The line you linked is the correct one, yes. `None` is `False` in Python in a boolean evaluation, yes. 

Yes we need both for valid requests but conversely I think that means that if either one of them, image file or play file couldn't be processed then we assume something went wrong and we shouldn't proceed. We don't need both to error out, one is enough, I think. Did I make a logical mistake?

This is of course quite rudimentary proper error propagation & handling should be a focus in the next release

Reply:
> The line you linked is the correct one, yes. None is False in Python in a boolean evaluation, yes.

Cool!

> Yes we need both for valid requests but conversely I think that means that if either one of them, image file or play file couldn't be processed then we assume something went wrong and we shouldn't proceed. We don't need both to error out, one is enough, I think. Did I make a logical mistake?

Yes, we need to return an error if either `image_file_name` or `item.play_file_name` is `None` (`False`). But in oder to satisfy `not (image_file_name or item.play_file_name)` you need both to be `False`.  If we express it with De Morgan's law, we need here `(not A) or (not B)` as the condition of the error which is equivalent with `not (A and B)`. But maybe I'm wrong again :)

Reply:
No you are correct! Thanks~~~